src/ui/public/visualize/visualize_legend.html

```diff
@@ -18,6 +21,10 @@
 
       <div class="legend-value-container">
         <div
+          kbn-accessible-click
+          data-label="{{legendData.label}}"
```

Comment:
What does this attribute do?

Reply:
Currently, the label doesn't indicate to the screen reader what will happen if the user clicks the button. Is there an elegant way to add an invisible label that says something like "Click for filtering options"? Maybe we can associate it with `aria-describedby`? Do you think this will be helpful?

Also, if the label is a value like "181", then the screen reader will just read this number aloud, which doesn't provide enough context for me to make sense of it. What if we added a noun to the end of this value? For example, if the aggregation was on "bytes", we could read aloud:

```
{{ `${legendData.label} + ${legendData.values.aggConfig.params.field.name}` }}
```

Which would cause the screen reader to announce "181 bytes". What do you think?

Reply:
The attribute itself is required by the handlers in the visualization, that are doing the `highlight` and `unhighlight` to find the actual value. Not ideal imho, but that's the way it works atm.

I will add a description for what the label triggering does, that's a very good idea.

I think extending the label with the fieldname, won't always work as smooth as it did in your example :-) So I would rather add this to the description like: "Filtering bytes fields for this value"

Reply:
I added it to the label instead, so you really can't miss it. I didn't add `aria-expanded` since that would require we can reference the legend options via `aria-controls`. But they use `ng-if` and not `ng-show` at the moment, so they are really removed from DOM and we cannot reference them when hidden. I also don't want to use `ng-show` in that place, since imho the options contain a significant amount of DOM nodes and having a dashboard with multiple visualizations on it, which several legend items, would render a significantly higher amount of DOM nodes with `ng-show`.

So I decided to not use `aria-expanded` in that case.